azurerm/internal/services/compute/virtual_machine.go

```diff
@@ -283,6 +283,98 @@ func virtualMachineOSDiskSchema() *schema.Schema {
 	}
 }
 
+func virtualMachineDataDiskSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Computed: true,
+		Elem: &schema.Resource{
```

Comment:
Can we have no data disk? then would we need to have ConfigModeAttr here so the user can remove all?

Reply:
Dropped `Computed` - since we can't currently support mixing in-line with using the attachment resource, so this should be fine.